### PR TITLE
Small-ish xenomorph refactor, hunter and sentinel rebalance

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

- Refactors a bunch of random xeno code a bit, fixes xenos runtiming literally every time they click something out of their range
- Reworked intents. You can hug xenos, grab them with grab intent and punch them on disarm/harm intent.
- Hunter slowed down by 50%
- Hunter health 125 => 200
- Sentinel health 150 => 225

## Why It's Good For The Game

Hunter is not fun to fight when that fast. This keeps them still considerably faster than a regular crew memeber, but at a more sane level. Sentinel hp buffed so they actually have some use and are not just an inferior hunter.

Code being better is also nice.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

It's a lot of changes, so it's kinda tough to test. Might wanna do a test merge first. I did check the move speed and basic xeno abilities work, though.

</details>

## Changelog
:cl:
balance: slowed down hunters while buffing hunter's and sentinels' health
refactor: refactored some xeno code
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
What did you change about disarming? Document the full extent of the changes, or write that there should be no mechanical differences.

**Comment by ivanmixo** [comment]
> What did you change about disarming? Document the full extent of the changes, or write that there should be no mechanical differences.

Did, sorry. This is so inconsequential I didn't really think it was worth really explaining. (when do you click on a xeno unarmed, wew.)

File: code/modules/mob/living/carbon/alien/alien.dm [hunk 1]
```diff
@@ -97,31 +91,31 @@ Proc: AddInfectionImages()
...
-	if (client)
-		for (var/i in GLOB.mob_living_list)
-			var/mob/living/L = i
-			if(HAS_TRAIT(L, TRAIT_XENO_HOST))
-				var/obj/item/organ/body_egg/alien_embryo/A = L.getorgan(/obj/item/organ/body_egg/alien_embryo)
-				if(A)
-					var/I = image('icons/mob/alien.dmi', loc = L, icon_state = "infected[A.stage]")
-					client.images += I
-	return
+	if(client)
```
Comment (Sarchutar) [inline code/modules/mob/living/carbon/alien/alien.dm]: did you forget to put a `!` here? like `if(!client)`

Comment (ivanmixo) [reply code/modules/mob/living/carbon/alien/alien.dm]:
```suggestion
	if(!client)
```

Possibly.

File: code/modules/mob/living/carbon/alien/humanoid/humanoid.dm [hunk 2]
```diff
@@ -65,10 +64,10 @@ GLOBAL_LIST_INIT(strippable_alien_humanoid_items, create_strippable_list(list(
 
 /mob/living/carbon/alien/humanoid/alien_evolve(mob/living/carbon/alien/humanoid/new_xeno)
 	drop_all_held_items()
-	..()
+	return ..()
 
 //For alien evolution/promotion/queen finder procs. Checks for an active alien of that type
-/proc/get_alien_type(var/alienpath)
+/proc/get_alien_type_in_hive(datum/alienpath)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/alien/humanoid/humanoid.dm:70]: Should this ensure that the alien belongs to the same hive? For example its not just some random xenobio spawn

**Review by PowerfulBacon — APPROVED** [review]
Seems fine

File: code/modules/mob/living/carbon/alien/humanoid/humanoid.dm (hunk 2 above)
Comment (ivanmixo) [reply code/modules/mob/living/carbon/alien/humanoid/humanoid.dm:70]: I don't think we support multiple hives unfortunately

**Pull request merged**